### PR TITLE
Simplify TLS for Knative and Kuebrenets + Document

### LINKED ISSUE #157: Kubernetes support no longer works out of the box with 0.4
Kubernetes support no longer works out of the box with 0.4

Auth added in version 0.4 relies on the secret being available
1. Yaml refers to the cert secret though Kubernetes default should be NoAuth
2. No documentation to move to TLS + Auth, no way to create the cert Secret

### PR DESCRIPTION
**Pull request by davidhadas**
Fixes #157 

## Changes the defaults: 
- Use an AUTH token
- Use TLS
- Verify Server on TLS is ROOT_CA is available
 
This simplifies much of the scripting and setups we need to do for both Knative and Kubernetes modes.

## ROOT_CA
ROOT_CA is now taken from a mounted secret, this should work in line with the Knative secrets. 
For some reason, the `internal-encryption=true` does not seem to work properly, hence cant test that ROOT_CA is loaded and works on Knative with `internal-encryption=true`, but it does work when manually creating the secrets and in Kubernetes. 

The design is that when `internal-encryption=true`, ROOT_CA will be available for `guard-gate` hence server verification will be made.  If no ROOT_CA, we use TLS without server verification (still better than HTTP as was done so far).

Meanwhile, removed e2e testing for the case with ROOT_CA, waiting to get `internal-encryption=true` to work and then test with this option enabled.

## Add Kind support
 - Added a simple, example of installing Guard in Kubernetes `./hack/kind/deplotKind` should work out of the box when kind and kubectl are installed. 
 
## Added documentation for Kubernetes installation 

**Release Note**
```release-note
Guard documentation and scripts were added to enable installation in Kind as Kubernetes example
Guard setup is now secured by default (AUTH token and TLS switched on) 
```

### CODE REVIEW

**Comment by davidhadas** [comment]
Failure in Security / analyze / Analyze (pull_request) test seems unrelated to the changes

**Comment by davidhadas** [comment]
> Failure in Security / analyze / Analyze (pull_request) test seems unrelated to the changes

My bad - the symbolic links used absolute-path instead of a relative one... :)

**Comment by psschwei** [comment]
Will take a look next week, maybe @maximilien can review if you need it sooner...

**Comment by davidhadas** [comment]
What should work out of the box:

- Running `./hack/kind/deployDevOnly.sh` (deploying without TLS and AUTH) 
- Running  `./hack/kind/deployProduction.sh` (deploying with TLS and AUTH)

In both cases, after you run the script you should have a working helloworld (and myapp) secured by guard on your kind.

To test that helloworld work you can use curl http://127.0.0.1/hello
To see guard works ok, you can `kubectl logs deployment/helloworld guard-rproxy` and see that:
1. Guard was started
2. It uses or does not use TLS and AUTH
3. It uses is able to communicate with the guard-service

Unfortunately learning now takes ~ 5-6 min, so you have to wait if you want to really test guard in action. 
An upcoming PR #176 is going to improve that and you will see learning done after ~ 1 minute.

**Comment by psschwei** [comment]
@davidhadas did you change your email or anything like that? CLA check is failing for some reason...

**Comment by psschwei** [comment]
and now it works :shrug:

**Comment by davidhadas** [comment]
> @davidhadas did you change your email or anything like that? CLA check is failing for some reason...

No - did not change email
Not sure why that is. 
It happened before and closing the issue and opening it fixed it. 
Let me try again to force it to check

**Comment by davidhadas** [comment]
image
